pkg/controller/framework_test.go

```diff
@@ -297,6 +302,13 @@ func runTests(t *testing.T, handlerFactory handlerFactory, tests []testCase) {
 		if test.additionalCheck != nil {
 			test.additionalCheck(t, test)
 		}
+		// makesure all the csi calls were executed.
+		if csiConnection.index < len(csiConnection.calls) {
```

Comment:
Is it guaranteed that the test will run the reconciler loop exactly the same number of times?

Reply:
I think yes. Otherwise, I think we need to fix such tests.